core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -1466,6 +1466,21 @@ sealed trait ZIO[-R, +E, +A] extends Serializable with ZIOPlatformSpecific[R, E,
   final def raceFirst[R1 <: R, E1 >: E, A1 >: A](that: ZIO[R1, E1, A1]): ZIO[R1, E1, A1] =
     (self.run race that.run).flatMap(ZIO.done(_)).refailWithTrace
 
+  /**
+   * Returns an effect that races this effect with all the specified effects,
+   * yielding the first result to complete, whether by success or failure. If
+   * neither effect completes, then the composed effect will not complete.
+   *
+   * WARNING: The raced effect will safely interrupt the "losers", but will not
+   * resume until the losers has been cleanly terminated. If early return is
+   * desired, then instead of performing `l raceAllFirst rs`, perform
+   * `l.disconnect raceAllFirst rs.flatMap(_.disconnect)`, which disconnects
+   * left and rights interrupt signal, allowing a fast return, with interruption
+   * performed in the background.
+   */
+  final def raceAllFirst[R1 <: R, E1 >: E, A1 >: A](ios: Iterable[ZIO[R1, E1, A1]]): ZIO[R1, E1, A1] =
```

Comment:
I think I would just call this `raceFirst` and make it an operator on the `ZIO` companion object instead of the `ZIO` trait but otherwise looks great!

Reply:
Sure, will do, I thought keeping it next to the other variants would make more sense, but I can move it to the companion.

Reply:
Well I think we already have a version of `raceAll` on the companion object so I think it will be in good company there'd I think when we have these variants that take a whole collection rather than a single value it is more idiomatic to have them on the companion object though I realize we haven't always done that in the past.